org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/execution/UpdateExecutionPageJob.java

```diff
@@ -34,15 +33,18 @@
 
     private static final int REPEAT_DELAY = 100;
 
-    private final Map<OperationDescriptor, OperationItem> operationItems;
+    private final Set<OperationItem> operationItems;
+    private final Set<OperationItem> removeableItems;
+
     private volatile boolean running;
     private final TreeViewer treeViewer;
 
     public UpdateExecutionPageJob(TreeViewer treeViewer) {
         super("Updating duration of non-finished operations");
 
         this.treeViewer = treeViewer;
-        this.operationItems = Maps.newHashMap();
+        this.operationItems = Sets.newConcurrentHashSet();
```

Comment:
Does it really cause overhead if you only synchronize the access to the two sets? I.e. keep the UI update out of the synchronized block?